app/components/combined_system_message/combined_system_message.js

```diff
@@ -0,0 +1,350 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {Text} from 'react-native';
+import {intlShape} from 'react-intl';
+
+import {Posts} from 'mattermost-redux/constants';
+import {displayUsername} from 'mattermost-redux/utils/user_utils';
+
+import CustomPropTypes from 'app/constants/custom_prop_types';
+
+import {makeStyleSheetFromTheme} from 'app/utils/theme';
+
+import FormattedText from 'app/components/formatted_text';
+
+import LastUsers from './last_users';
+
+const {
+    JOIN_CHANNEL, ADD_TO_CHANNEL, REMOVE_FROM_CHANNEL, LEAVE_CHANNEL,
+    JOIN_TEAM, ADD_TO_TEAM, REMOVE_FROM_TEAM, LEAVE_TEAM,
+} = Posts.POST_TYPES;
+
+const postTypeMessage = {
+    [JOIN_CHANNEL]: {
+        one: {
+            id: ['combined_system_message.first_user', 'combined_system_message.joined_channel'],
```

Comment:
Do these need entries in en.json? Also, do they need `mobile.combined_system_message` as a prefix?

Reply:
Woah, I swear I've added all those locale strings in en.json but maybe was not committed. I'll add those for sure, 0/5 on prefix but I will just add it.

Reply:
I think we might need the prefix otherwise the strings will get paved over then the translations are next updated from the web app